crates/driver/src/domain/competition/auction.rs

```diff
@@ -605,3 +637,66 @@ pub enum Error {
     #[error("blockchain error: {0:?}")]
     Blockchain(#[from] blockchain::Error),
 }
+
+trait OrderingKey: Send + Sync + 'static {
+    type Key: Ord + Send + Sync + 'static;
+
+    fn key(&self, order: &order::Order, tokens: &Tokens, solver: &eth::H160) -> Self::Key;
+
+    /// Returns a comparator that compares two orders based on the key in
+    /// reverse order.
+    fn comparator(self: Arc<Self>) -> OrderComparator {
+        Arc::new(
+            move |a: &order::Order, b: &order::Order, tokens: &Tokens, solver: &eth::H160| {
+                self.key(a, tokens, solver)
+                    .cmp(&self.key(b, tokens, solver))
+                    .reverse()
+            },
+        )
+    }
+}
+
+struct OrderClass;
+impl OrderingKey for OrderClass {
+    type Key = i32;
+
+    // Market orders are preferred over limit orders, as the expectation is that
+    // they should be immediately fulfillable. Liquidity orders come last, as they
+    // are the most niche and rarely used.
+    fn key(&self, order: &order::Order, _tokens: &Tokens, _solver: &eth::H160) -> Self::Key {
+        match order.kind {
+            order::Kind::Market => 2,
+            order::Kind::Limit { .. } => 1,
+            order::Kind::Liquidity => 0,
+        }
+    }
+}
+
+struct ExternalPrice;
+impl OrderingKey for ExternalPrice {
+    type Key = num::BigRational;
+
+    fn key(&self, order: &order::Order, tokens: &Tokens, _solver: &eth::H160) -> Self::Key {
+        order.likelihood(tokens)
+    }
+}
+
+struct CreationTimestamp;
+impl OrderingKey for CreationTimestamp {
+    type Key = Option<util::Timestamp>;
+
+    fn key(&self, order: &order::Order, _tokens: &Tokens, _solver: &eth::H160) -> Self::Key {
+        (order.created.0
+            > u32::try_from((Utc::now() - Duration::minutes(2)).timestamp()).unwrap_or(u32::MAX))
```

Comment:
why the 2 minutes lifetime? please could you document it / potentially make it as its own const?

Reply:
> why the 2 minutes lifetime?

The reason is specified in both the PR description and the linked issue.

> please could you document it / potentially make it as its own const?

Done

Reply:
> The reason is specified in both the PR description and the linked issue.

Yeah, but my question is why 2 minutes exactly? why not 3, or 1? is it just to "try" and see / gut feeling? my concern with this is, if we discover that 2 minutes is not enough for whatever reason, to change this, implies a full release since it is hardcoded. That's why I am asking :thinking: